### DISCUSSION TITLE
Not successful to make it working

### DISCUSSION THREAD

**Discussion by dariusz737** [dzurikmiroslav/esp32-evse#58] [General]
What is authorization?
I have such errors
All meters are off in board.config. Fixed cable.
Please help.

<img width="713" alt="Zrzut ekranu 2024-07-24 o 21 54 24" src="https://github.com/user-attachments/assets/8f665841-8391-4c12-b8e5-011c04851728">
<img width="382" alt="Zrzut ekranu 2024-07-24 o 21 54 46" src="https://github.com/user-attachments/assets/0bae0544-21b7-460e-8f5f-2c02dc28aa6e">

**Comment by dzurikmiroslav** [comment]
In your log I see "Error bits 2" which has value https://github.com/dzurikmiroslav/esp32-evse/blob/master/components/evse/include/evse.h#L12 
Diode short mean problem of diode in EV side, are you testing with car? or have EV simulator circuit like this?
![image](https://github.com/user-attachments/assets/ea57b96b-ec0e-4c34-9bda-93350710aa7c)
Do you have a oscilloscope ? Check if CP go to -12V.

**Comment by dariusz737** [comment]
Thank you Miroslav for prompt answer. I did everything described in wiki, (with simulator), all values are correct. After connecting car when charging is not "enabled", green diode starts to blink. But after I switched "enabled", it shows error. I have direct connection, no socket. Is there (should be) any resistor build  in charging cable? Is there an option to have "disable" charging by default after restart? I don't have oscilloscope :(  Please explain me what is "authorisation" and what is expected behaviour when sen to ON or OFF.

**Comment by dzurikmiroslav** [comment]
When Authorization is ON, you always need to click Authorize button after connecting EV to start charging process.

Set in settings System / Charging available -> OFF and connect EV simulator in state A (diode + 2k72 resistor only).
CP should do only negative signal, measure voltage on CP.

**Reply by nagyrobi** [reply]
> 2k72 resistor

Was about to ask: the drawing shows a resistor of 2k74, here you mention 2k72, but I assume these are just mistypes, I guess we're talking just standard 2.7k resistors... (1% maybe).

**Reply by dzurikmiroslav** [reply]
Sorry, yes I mean 2k74. I think 2k7 should be suitable, I can see these being used on the internet...

**Comment by dariusz737** [comment]
Problem was with TL081, I have changed it and started working. But, if I chose cur_vlt, there is an error with reading current, and immediately disconnect with Error bits 1.  Most probably reading much to much current. When CT is noc connected, shows 0.32A, and correct voltage reading, allowing to charge. Any chance to see current reading without charging, for example from some test circuit. CT is 2000/1 ...

**Reply by dzurikmiroslav** [reply]
I encountered that when too high a voltage is applied to single ADC1 input, all other ADC1 inputs can't be measured accurately either. Of course, exceeding the maximum voltage can permanently damage the ESP32 (I also have one that seems to work, but not all inputs work correctly and it produces a lot of heat).
This problem occurs if you select the electricity meter only 'cur' ? Could the overvoltage be caused by the AC voltage sensing?

The surest thing would be to measure the voltage on the ESP32 inputs with an oscilloscope...

**Reply by dariusz737** [reply]
Correct voltage readings.  I can measure anything in circuit because it disconnects in split of the second after connecting. Actually what is strange, it disconnects even when car didn't even started charging and there is no current flow, or it is marginal. What does it mean ENERGY_METER_THREE_PHASES=n ?  If I measure 1 phase, does software multiplay readings by 3 ? Should I then leave ADC channels set for not used GPIO, or put "n" there?

**Reply by dzurikmiroslav** [reply]
Config property ENERGY_METER_THREE_PHASES says if there is 1 or 3 current/voltage sensing ADC channels on board.
Is you set ENERGY_METER_THREE_PHASES=n and set EVSE setting Energy meter / Type to Current sensing/Current and voltage sensing, you will be able to measure only single phase consumption. And yes ENERGY_METER_L2_CUR_ADC_CHANNEL, ENERGY_METER_L2_VLT_ADC_CHANNEL etc. will be ignored, you can leave empty value.

When you need record the estimated power consumption select Energy meter / Type to Dummy single/three phase (consumption will be calculated from provided voltage and maximum current).

Board with only single phase energy meter I planned for portable chargers on schuko socket...

**Reply by dariusz737** [reply]
I have solved problem with current reading (my mistake). Instead of 22ohm, I have put 22kohm :D  But please, for you it is 1 minute of coding :) . Make option to multiply 1 phase reading by 3. My cars make perfect balanced load on three phases, it's pointless to measure 3 phases. And would be absolutely flawless to have native HA integration with just four items, connection status, NO-OFF charging, charging current setting, and power meter readings. ;)  I there any option for push notification, from EVSE instead ofsending data or inquery  on preselected interval.

**Reply by dzurikmiroslav** [reply]
I know it :-) I'm too lazy to learn resistor color codes and prefer to use a multi-meter.
I make new version 0.9.0, which has three phases option in energy meter settings.

Yes, HA integration would be great, but I haven't make anything like that yet. I don't know HA very well, is the REST api for integration not enough?

**Reply by nagyrobi** [reply]
I plan to integrate it with [ESPHome via ModBUS](https://github.com/dzurikmiroslav/esp32-evse/discussions/59).

**Reply by nagyrobi** [reply]
> I make new version 0.9.0, which has three phases option in energy meter settings.

Does this change Modbus register 315? And the description on [Connection examples](https://github.com/dzurikmiroslav/esp32-evse/wiki/ESP32-S2-DA#connection-examples)?

**Reply by dariusz737** [reply]
I see option to switch 3 phases now in settings. The point is that when 1 phase metering selected, it should multiply power by 3 assuming balanced load on 3 phases. Please add such option. Where are you from exactly in Slovakia, maybe not so far so we can meet for a big beer :D

**Reply by dzurikmiroslav** [reply]
> > I make new version 0.9.0, which has three phases option in energy meter settings.
> 
> Does this change Modbus register 315? And the description on [Connection examples](https://github.com/dzurikmiroslav/esp32-evse/wiki/ESP32-S2-DA#connection-examples)?

Yes, register 315 values are now different, I will update docs...

**Reply by dzurikmiroslav** [reply]
> I see option to switch 3 phases now in settings. The point is that when 1 phase metering selected, it should multiply power by 3 assuming balanced load on 3 phases. Please add such option. Where are you from exactly in Slovakia, maybe not so far so we can meet for a big beer :D

Yes, when you have board config with single phase energy meter, and in EVSE / Energy meter settings three phases are true, L2&L3 voltages and currents is copy of L1. I am from central Slovakia ;-)
